doc/GUIDE.md

```diff
@@ -103,94 +99,33 @@ michael@d30748af6d3d:~$ stack new helloworld new-template
-message that we are missing the right GHC version.
-
-Let's run stack setup:
-
-```
-michael@d30748af6d3d:~/helloworld$ stack setup
-Downloaded ghc-7.10.2.
-Installed GHC.
-stack will use a sandboxed GHC it installed
-For more information on paths, see 'stack path' and 'stack exec env'
-To use this GHC and packages outside of a project, consider using:
-stack ghc, stack ghci, stack runghc, or stack exec
-```
-
-It doesn't come through in the output here, but you'll get intermediate
-download percentage statistics while the download is occurring. This command
-may take some time, depending on download speeds.
-
-__NOTE__: GHC will be installed to your global stack root directory, so
-calling `ghc` on the command line won't work. See the `stack exec`,
-`stack ghc`, and `stack runghc` commands below for more information.
-
 ### stack build
 
 Next, we'll run the most important stack command: `stack build`.
 
-__NOTE__: If you forgot to run `stack setup` in the previous step you'll get an
-error:
-
 ```
-michael@d30748af6d3d:~$ cd helloworld/
+michael@d30748af6d3d:~$ cd helloworld
 michael@d30748af6d3d:~/helloworld$ stack build
-No GHC found, expected version 7.10.2 (x86_64) (based on resolver setting in /home/michael/helloworld/stack.yaml).
-Try running stack setup
+# installing ... building ...
 ```
 
-stack needs GHC in order to build your project, and `stack setup` must be run to
-check whether GHC is available (and install it if not).
+stack needs a GHC in order to build your project. stack will discover that you
+are missing it and will install it for you. You can do this manually by using
+the `stack setup` command.
 
-Having run `stack setup` successfully, `stack build` should build our project:
+You'll get intermediate download percentage statistics while the download is
+occurring. This command may take some time, depending on download speeds.
 
-```
-michael@d30748af6d3d:~/helloworld$ stack build
-helloworld-0.1.0.0: configure
-Configuring helloworld-0.1.0.0...
-helloworld-0.1.0.0: build
-Preprocessing library helloworld-0.1.0.0...
-[1 of 1] Compiling Lib              ( src/Lib.hs, .stack-work/dist/x86_64-linux/Cabal-1.22.4.0/build/Lib.o )
-In-place registering helloworld-0.1.0.0...
-Preprocessing executable 'helloworld-exe' for helloworld-0.1.0.0...
-[1 of 1] Compiling Main             ( app/Main.hs, .stack-work/dist/x86_64-linux/Cabal-1.22.4.0/build/helloworld-exe/helloworld-exe-tmp/Main.o )
-Linking .stack-work/dist/x86_64-linux/Cabal-1.22.4.0/build/helloworld-exe/helloworld-exe ...
-helloworld-0.1.0.0: install
-Installing library in
-/home/michael/helloworld/.stack-work/install/x86_64-linux/lts-3.2/7.10.2/lib/x86_64-linux-ghc-7.10.2/helloworld-0.1.0.0-6urpPe0MO7OHasGCFSyIAT
-Installing executable(s) in
-/home/michael/helloworld/.stack-work/install/x86_64-linux/lts-3.2/7.10.2/bin
-Registering helloworld-0.1.0.0...
-```
+__NOTE__: GHC will be installed to your global stack root directory, so
```

Comment:
Can't think of how better to word this part, but the "global stack root directory" seems not the best wording to me for a beginner. Maybe something about "GHC will be installed to an internal stack location," or maybe just remove the word "global" ? Or do we call STACK_ROOT or whatever it is the "global stack root" elsewhere?

I might just be being picky here, feel free to ignore, haha.